pre_commit/commands/run.py

```diff
@@ -175,10 +177,15 @@ def _run_single_hook(
         duration = round(time.time() - time_before, 2) or 0
         diff_after = _get_diff()
 
-        # if the hook makes changes, fail the commit
+        # if the hook makes changes, fail the commit or add the changes automatically
         files_modified = diff_before != diff_after
+        hook_failed = bool(retcode) or (files_modified and not hook.commit_changes)
 
-        if retcode or files_modified:
+        if files_modified and hook.commit_changes:
+            git.update_changes()
+            diff_after = b''
```

Comment:
Looking at the line above (where `files_modified` is calculated), should this be `diff_before` instead of the empty byte-string?

Reply:
Thinking about this again, as written this doesn't interact correctly with other hooks that don't commit changes, depending on run order.

Right now, if a hook with `commit_changes=True` runs after a hook with `commit_changes=False` that makes modifications, it will add both modifications to the index (and then will not commit, because the `commit_changes=False` hook will have failed).  I don't think this is the correct behavior.  Having a hook add precisely its modifications to the index seems doable, but probably somewhat complicated. My inclination here is to just not add anything to the index in this case -- does that seem reasonable?

Reply:
I wonder if it would be weird to just do `diff_after = _get_diff()` again 🤔 It *should* be the empty string if `update_changes` worked correctly. Probably doesn't matter either way; this is fine as is 👍

Reply:
Did that + added an assert.